pgd_splicer/ProcessPDBTask.py

```diff
@@ -403,6 +403,58 @@ def atom_noamino(s):
     return s.startswith("ATOM ") and not amino_present(s)
 
 
+class OccupancySelect(Select):
+    """
+    This class implements the occupancy awareness selection.
+
+    Each disordered residue contains one or more disordered atom structures.  These structures contain one or more atoms, each with its own altloc and occupancy values.  As each atom associated with a particular altloc has its own occupancy value, the altloc with the highest average occupancy value is identified.  Atoms with other altloc values are removed from the structure.
+    """
+
+    best_atoms = {}
+
+    def accept_residue(self, residue):
+        """
+        This method is being overloaded to perform the calculations required to properly accept atoms in the model based on the highest average occupancy value per altloc.
+        """
+
+        if residue.is_disordered():
+            res_occ = {}
+            for atom in residue:
+                name = atom.name
+                res_occ[name] = {}
+                if atom.is_disordered():
+                    altloc = atom.get_altloc()
+                    occ = atom.get_occupancy()
+                    try:
+                        res_occ[name][altloc].append(occ)
+                    except KeyError:
+                        res_occ[name][altloc] = [occ]
+
+            occ_list = {}
+            for occ_vals in [res_occ[k] for k in res_occ if res_occ[k]]:
+                for altloc in occ_vals:
+                    for occ in occ_vals[altloc]:
+                        try:
+                            occ_list[altloc].append(occ)
+                        except KeyError:
+                            occ_list[altloc] = [occ]
+
+            best_altlocs = sorted(occ_list, key=lambda k: sum(occ_list[k])/len(occ_list[k]), reverse=True)
+            self.best_atoms[residue] = {atom: next(altloc for altloc in best_altlocs if altloc in res_occ[atom] or ' ') for atom in res_occ}
```

Comment:
These lines, if you can, should also be limited to 80 characters. If it breaks readability then don't bother.